impl/src/main/java/org/ehcache/internal/store/disk/DiskStore.java

```diff
@@ -484,6 +491,22 @@ public void disableStoreEventNotifications() {
     return new DiskStoreIterator();
   }
 
+  @Override
+  public ValueHolder<V> getAndFault(K key) throws CacheAccessException {
+    return get(key, true);
+  }
+
+  @Override
+  public ValueHolder<V> computeIfAbsentAndFault(K key, Function<? super K, ? extends V> mappingFunction) throws CacheAccessException {
+    return internalComputeIfAbsent(key, mappingFunction, true);
+  }
+
+  @Override
+  public boolean flush(K key, ValueHolder<V> valueHolder, CachingTier<K, V> cachingTier) {
+    int hash = hash(key.hashCode());
+    return segmentFor(hash).flush(key, hash, (DiskValueHolder<V>) valueHolder, cachingTier);
```

Comment:
it's actually a mistake and a bug (that also probably indicates a lack in the test coverage): this should be a ValueHolder of the CachingTier, not of the authoritative tier.

That's the reason why there also is a CachingTier object passed in: the ValueHolder interface doesn't expose expiration methods that are needed to copy the expiration back to the authoritative tier, hence the authoritative tier asks the caching tier for that data to copy it in its own ValueHolder.

We may want to revisit the ValueHolder interface, or at least the way I implemented this expiration-copying mechanism.